target/snitch_cluster/sw/snax/streamer-gemm-conv/src/snax-streamer-gemm-conv-lib.c

```diff
@@ -0,0 +1,139 @@
+                      K * Fy * Fx * C * sizeof(int8_t));
+}
+
+// Set STREAMER configuration CSR
+void set_conv_streamer_csr(
+    int Aslstride0, int Aslstride1, int Atlbound0, int Atlstride0,
+    int Atlbound1, int Atlstride1, int Atlbound2, int Atlstride2, int Atlbound3,
+    int Atlstride3, int Atlbound4, int Atlstride4, int Atlbound5,
+    int Atlstride5, int Bslstride0, int Bslstride1, int Btlbound0,
+    int Btlstride0, int Btlbound1, int Btlstride1, int Btlbound2,
+    int Btlstride2, int Cslstride0, int Cslstride1, int Ctlbound0,
+    int Ctlstride0, int Ctlbound1, int Ctlstride1, int Ctlbound2,
+    int Ctlstride2, int delta_local_a, int delta_local_b, int delta_local_c) {
+    // loop bounds, from innermost to outermost, for data mover A
+    write_csr(960, Atlbound0);
+    write_csr(961, Atlbound1);
+    write_csr(962, Atlbound2);
+    write_csr(963, Atlbound3);
+    write_csr(964, Atlbound4);
+    write_csr(965, Atlbound5);
+
+    // loop bounds, from innermost to outermost, for data mover B
+    write_csr(966, Btlbound0);
+    write_csr(967, Btlbound1);
+    write_csr(968, Btlbound2);
+
+    // loop bounds, from innermost to outermost, for data mover C
+    write_csr(969, Ctlbound0);
+    write_csr(970, Ctlbound1);
+    write_csr(971, Ctlbound2);
+
+    // temporal strides for A
+    write_csr(972, Atlstride0);
+    write_csr(973, Atlstride1);
+    write_csr(974, Atlstride2);
+    write_csr(975, Atlstride3);
+    write_csr(976, Atlstride4);
+    write_csr(977, Atlstride5);
+
+    // temporal strides for B
+    write_csr(978, Btlstride0);
+    write_csr(979, Btlstride1);
+    write_csr(980, Btlstride2);
+
+    // temporal strides for C
+    write_csr(981, Ctlstride0);
+    write_csr(982, Ctlstride1);
+    write_csr(983, Ctlstride2);
+
+    // spatial strides for A
+    write_csr(984, Aslstride0);
+    write_csr(985, Aslstride1);
+
+    // spatial strides for B
+    write_csr(986, Bslstride0);
+    write_csr(987, Bslstride1);
+
+    // spatial strides for C
+    write_csr(988, Cslstride0);
+    write_csr(989, Cslstride1);
+
+    // base ptr for A
+    write_csr(990, (uint32_t)(delta_local_a + snrt_l1_next()));
+
+    // base ptr for B
+    write_csr(991, (uint32_t)(delta_local_b + snrt_l1_next()));
+
+    // base ptr for C
+    write_csr(992, (uint32_t)(delta_local_c + snrt_l1_next()));
+}
+
+// Set CSR to start STREAMER
+void set_conv_streamer_start() { write_csr(993, 1); }
+
+// Set GEMM configuration CSR
+void set_conv_block_gemm_csr(int tempLoop0, int tempLoop1, int tempLoop2,
+                             int subtractions) {
+    // set loop bounds, from innermost to outermost, aka from K to N to M
+    write_csr(995, tempLoop0);
+    write_csr(996, tempLoop1);
+    write_csr(997, tempLoop2);
+
+    // set subtraction a and b
+    write_csr(998, subtractions);
+}
+
+// Set CSR to start GEMM
+void set_conv_block_gemm_start() { write_csr(999, 1); }
+
+// Poll until Streamer and GEMM accelerator finish
```

Comment:
Are these polling though?
```suggestion
// Stall until Streamer and GEMM accelerator finish
```